pkg/api/dashboard.go

```diff
@@ -440,26 +436,21 @@ func (hs *HTTPServer) postDashboard(c *contextmodel.ReqContext, cmd dashboards.S
 		Overwrite: cmd.Overwrite,
 	}
 
-	dashboard, err := hs.DashboardService.SaveDashboard(ctx, dashItem, allowUiUpdate)
+	dashboard, saveErr := hs.DashboardService.SaveDashboard(ctx, dashItem, allowUiUpdate)
 
 	if hs.Live != nil {
 		// Tell everyone listening that the dashboard changed
 		if dashboard == nil {
 			dashboard = dash // the original request
 		}
 
-		userDTODisplay, err := user.NewUserDisplayDTOFromRequester(c.SignedInUser)
```

Comment:
This was overwriting `err` with `nil` in all cases, breaking error handling further down. I renamed  `err` to `saveErr` above to clearly show where it is being used since the logic is a bit inscrutable.